app/models/manageiq/providers/embedded_ansible/automation_manager/playbook.rb

```diff
@@ -1,5 +1,159 @@
+    job = reload_job_status(job_id)
+    if job.status == 'successful'
+      @task.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_OK, "Playbook launched successfully")
+    elsif job.status == 'failed'
+      @task.update_status(MiqTask::STATE_FINISHED, MiqTask::STATUS_ERROR, "[#{job_id}] Playbook launch failed")
+    else
+      @task.update_status(MiqTask::STATE_ACTIVE, MiqTask::STATUS_OK, "Playbook still running")
+      MiqQueue.put(
+        :class_name   => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook",
+        :method_name  => 'queue_status',
+        :role         => 'embedded_ansible',
+        :args         => [job.id, task.id],
+        :task_id      => nil,
+        :miq_callback => build_queue_callback(task),
+        :zone         => @tower.my_zone
+      )
+    end
+  end
+
+  def queue_runner(userid, options)
+    tower = manager
+
+    task = MiqTask.create(:name => "Running Ansible Playbook [#{manager_ref}]", :userid => userid, :context_data => options)
+    MiqQueue.put(
+      :class_name   => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook",
+      :method_name  => 'run_playbook',
+      :role         => 'embedded_ansible',
+      :args         => [task.id],
+      :miq_callback => build_queue_callback(task),
+      :zone         => tower.my_zone
+    )
+    task.id
+  end
+
+  def self.build_parameter_list(name, description, info)
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    playbook.build_parameter_list(name, description, info)
+  end
+  private_class_method :build_parameter_list
+
+  private
+
+  def build_queue_callback(task)
+    {
+      :class_name  => task.class.name,
+      :instance_id => task.id,
+      :method_name => :queue_callback_on_exceptions,
+      :args        => ['Finished']
+    }
+  end
+
+  def build_instance_vars(task)
+    @task = task
+    @options = task.context_data
+    @tower = manager
+    self
+  end
+
+  def build_parameter_list(template_name, description, info = nil)
+    options = info.nil? ? @options[:config_info] : info
+    tower = @tower || manager
+    params = {
+      :name                     => template_name,
+      :description              => description || '',
+      :project                  => configuration_script_source.manager_ref,
+      :playbook                 => name,
+      :become_enabled           => options[:become_enabled].present?,
+      :verbosity                => options[:verbosity].presence || 0,
+      :ask_variables_on_launch  => true,
+      :ask_limit_on_launch      => true,
+      :ask_inventory_on_launch  => true,
+      :ask_credential_on_launch => true
+    }
+
+    if options[:limit]
+      params[:limit] = options[:limit]
+    end
+    if options[:extra_vars]
+      params[:extra_vars] = options[:extra_vars].deep_stringify_keys!.to_json
+    end
+
+    hosts = options[:hosts] || options[:extra_vars][:hosts] || 'localhost'
```

Comment:
If `hosts` is `nil`, we should use `tower.provider.default_inventory` which is `localhost` already. Don't create a new inventory for `localhost`.